### DISCUSSION TITLE
Technology/Tooling: Changesets

### DISCUSSION THREAD

**Discussion by alexanderniebuhr** [astrolicious/astrolicious#10] [Board]
# Decision Document: Adopting Changesets in the Astrolicious

## Executive Summary

This document proposes the adoption of Changesets as the standard tool for managing versioning and publishing processes within the Astrolicious community's projects. Changesets is selected for its ability to automate and streamline release workflows, ensuring consistent and efficient handling of package versions and changelogs across our diverse project landscape.

## Background

The Astrolicious community is committed to maintaining a high standard of development practices, including the management of releases across multiple projects. As our community grows and the complexity of our projects increases, the need for a unified, efficient release workflow solution becomes paramount. After careful consideration, Changesets has been identified as the ideal tool to meet these requirements.

## Proposal

We propose integrating Changesets into our projects to manage versioning and publishing workflows. Changesets offers a suite of features designed to simplify the release process, including:

- Automated version management based on semantic versioning principles.
- Generation of comprehensive changelogs from pull request descriptions.
- Batched package releases to ensure dependencies within monorepos are updated together, maintaining internal consistency.

## Rationale

The choice of Changesets is based on several considerations:

- **Efficiency:** Changesets automates several aspects of the release process, reducing manual effort and minimizing the risk of errors.
- **Consistency:** By enforcing semantic versioning and generating standardized changelogs, Changesets ensures that all projects maintain consistent release practices.
- **Flexibility:** Changesets is designed to work well in monorepo setups, making it suitable for the Astrolicious community's diverse project structures.
- **Community Support:** Changesets is backed by a strong community and ongoing development support, ensuring that it remains a reliable and up-to-date solution for release management.

## Implementation Plan

The adoption of Changesets will be carried out in the following phases:

1. **Pilot Implementation:** Select a few projects within the community to integrate Changesets, allowing us to assess its impact and refine our approach.
2. **Gather Feedback:** Collect insights and feedback from the pilot projects to identify any challenges and best practices for broader implementation.
3. **Community-wide Rollout:** Gradually introduce Changesets across all applicable Astrolicious community projects, providing support and documentation to facilitate the transition.
4. **Continuous Monitoring and Improvement:** Regularly review the effectiveness of Changesets in our workflow, staying open to adjustments and optimizations as needed.

## Conclusion

Implementing Changesets as our standard release workflow solution represents a strategic enhancement to the Astrolicious community's development practices. By streamlining versioning and changelog management, we aim to improve efficiency, maintain high standards of consistency, and support our community's growth and innovation. We invite feedback and participation from all community members as we move forward with this implementation.

**Comment by florian-lefebvre** [comment]
I think changesets is essential. It would great to have a template to setup them properly for github usage (like on astro repos)
